AssemblerLib/PETScVectorMatrixBuilder.h

```diff
@@ -0,0 +1,62 @@
+/*!
+  \file PETScVectorMatrixBuilder.h
+  \author Wenqing Wang
+  \date   2014.11
+  \brief  Interface to create PETSc matrix and vector
+
+  \copyright
+  Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+             Distributed under a Modified BSD License.
+               See accompanying file LICENSE.txt or
+               http://www.opengeosys.org/project/license
+*/
+
+#ifndef ASSEMBLERLIB_PETSCVECTORMATRIXBUILDER_H_
+#define ASSEMBLERLIB_PETSCVECTORMATRIXBUILDER_H_
+
+#include <petscmat.h>
+
+#include "MathLib/LinAlg/PETSc/PETScMatrixOption.h"
+
+#include "AssemblerLib/MeshComponentMap.h"
+
+namespace AssemblerLib
+{
+
+template <typename MatrixType_, typename VectorType_>
+class PETScVectorMatrixBuilder
+{
+public:
+    typedef VectorType_ VectorType;
+    typedef MatrixType_ MatrixType;
+
+public:
+    /*!
+        \param vec_size       The size of the vector, either global or local
+        \param is_global_size The flag of the type of vec_size, i.e. whether it is a global size
+                              or local size. The default is true.
+    */
+    static
+    VectorType* createVector(const PetscInt vec_size, const bool is_global_size = true)
```

Comment:
In case the vector size is given locally for each sub-domain, then the global size is calculated in size the constructor . It is used when one want to partition vector manually, and it will be used late on. There are also tests about it in TestGlobalVectorInterface.cpp